cli/backup.go

```diff
@@ -187,39 +187,45 @@ func waitForBcpStatus(ctx context.Context, cn *pbm.PBM, bcpName string) (err err
 // changed to pbm.StatusError with respective error text emitted. It doesn't change meta on
 // storage nor in DB (backup is ok, it just doesn't cluster), it is just "in-flight" changes
 // in given `bcps`.
-func bcpsMatchCluster(bcps []pbm.BackupMeta, shards []pbm.Shard, confsrv string, mapRS pbm.RSMapFunc) {
-	sh := make(map[string]struct{}, len(shards))
+func bcpsMatchCluster(bcps []pbm.BackupMeta, shards []pbm.Shard, confsrv string, rsMap map[string]string) {
+	sh := make(map[string]bool, len(shards))
 	for _, s := range shards {
-		sh[s.RS] = struct{}{}
+		sh[s.RS] = s.RS == confsrv
 	}
 
-	var buf []string
```

Comment:
In case of a large amount of backup and/or replica sets in the cluster, it may have a noticeable impact on pbm status load time.
```
name                            old time/op    new time/op     delta
BcpMatchCluster3x10-12             335ns ± 7%     1348ns ± 8%     +302.15%  (p=0.029 n=4+4)
BcpMatchCluster3x100-12           2.86µs ± 9%    12.49µs ±14%     +337.23%  (p=0.029 n=4+4)
BcpMatchCluster17x100-12          22.3µs ± 7%     39.9µs ±13%      +78.93%  (p=0.029 n=4+4)
BcpMatchCluster3x1000-12          32.0µs ±12%    128.0µs ±16%     +300.21%  (p=0.029 n=4+4)
BcpMatchCluster1000x1000-12       28.9ms ± 6%     36.5ms ±18%      +26.28%  (p=0.029 n=4+4)
BcpMatchCluster3x10Err-12         59.0ns ± 8%     59.6ns ±20%         ~     (p=0.686 n=4+4)
BcpMatchCluster1000x1000Err-12    38.7µs ± 7%     43.5µs ±21%         ~     (p=0.114 n=4+4)

name                            old alloc/op   new alloc/op    delta
BcpMatchCluster3x10-12             0.00B         640.00B ± 0%        +Inf%  (p=0.029 n=4+4)
BcpMatchCluster3x100-12            0.00B        6400.00B ± 0%        +Inf%  (p=0.029 n=4+4)
BcpMatchCluster17x100-12            583B ± 0%      7047B ± 0%    +1109.31%  (p=0.029 n=4+4)
BcpMatchCluster3x1000-12           0.00B       64000.00B ± 0%        +Inf%  (p=0.029 n=4+4)
BcpMatchCluster1000x1000-12       41.0kB ± 0%    113.2kB ± 0%     +176.13%  (p=0.029 n=4+4)
BcpMatchCluster3x10Err-12          0.00B           0.00B              ~     (all equal)
BcpMatchCluster1000x1000Err-12    41.0kB ± 0%     49.2kB ± 0%      +19.99%  (p=0.029 n=4+4)

name                            old allocs/op  new allocs/op   delta
BcpMatchCluster3x10-12              0.00           20.00 ± 0%        +Inf%  (p=0.029 n=4+4)
BcpMatchCluster3x100-12             0.00          200.00 ± 0%        +Inf%  (p=0.029 n=4+4)
BcpMatchCluster17x100-12            1.00 ± 0%     201.00 ± 0%   +20000.00%  (p=0.029 n=4+4)
BcpMatchCluster3x1000-12            0.00         2000.00 ± 0%        +Inf%  (p=0.029 n=4+4)
BcpMatchCluster1000x1000-12         2.00 ± 0%    2002.00 ± 0%  +100000.00%  (p=0.029 n=4+4)
BcpMatchCluster3x10Err-12           0.00            0.00              ~     (all equal)
BcpMatchCluster1000x1000Err-12      2.00 ± 0%       2.00 ± 0%         ~     (all equal)
```